core/src/main/resources/error/error-classes.json

```diff
@@ -780,8 +780,21 @@
   },
   "INVALID_SCHEMA" : {
     "message" : [
-      "The expression <expr> is not a valid schema string."
-    ]
+      "The input schema <inputSchema> is not a valid schema string."
+    ],
+    "subClass" : {
+      "PARSE_ERROR" : {
+        "message" : [
+          "Cannot parse the schema: ",
+          "<reason>"
+        ]
+      },
+      "UNEXPECTED_INPUT_TYPE" : {
```

Comment:
How about `NON_STRING_LITERAL`?

Reply:
renamed

Reply:
[af157d2](https://github.com/apache/spark/pull/38754/commits/af157d2bf34489ebbd116f6077eb13c5287971cc) rename ` INVALID_SCHEMA.UNEXPECTED_INPUT_TYPE` to `INVALID_SCHEMA.NON_STRING_LITERAL`